dask_expr/_rolling.py

```diff
@@ -0,0 +1,218 @@
+import functools
+from collections import namedtuple
+from numbers import Integral
+
+from dask.dataframe.dispatch import meta_nonempty
+
+from dask_expr._collection import new_collection
+from dask_expr._expr import Blockwise, Expr, Projection, make_meta
+
+BlockwiseDep = namedtuple(typename="BlockwiseDep", field_names=["iterable"])
+
+
+class RollingReduction(Expr):
+    _parameters = [
+        "frame",
+        "window",
+        "kwargs",
+        "how_args",
+        "how_kwargs",
+    ]
+    _defaults = {
+        "kwargs": None,
+        "how_args": (),
+        "how_kwargs": None,
+    }
+    how = None
+
+    @functools.cached_property
+    def npartitions(self):
+        return self.frame.npartitions
+
+    def _divisions(self):
+        return self.frame.divisions
+
+    @functools.cached_property
+    def _meta(self):
+        rolling = meta_nonempty(self.frame._meta).rolling(self.window, **self.kwargs)
+        meta = getattr(rolling, self.how)(*self.how_args, **self.how_kwargs or {})
+        return make_meta(meta)
+
+    @functools.cached_property
+    def kwargs(self):
+        return {} if self.operand("kwargs") is None else self.operand("kwargs")
+
+    def _simplify_up(self, parent):
+        if isinstance(parent, Projection):
+            return type(self)(self.frame[parent.operand("columns")], *self.operands[1:])
+
+    @property
+    def _has_single_partition(self):
```

Comment:
I don't like this name (not a blocker, can do in a follow up). It will let multi-partition frames through when window=1 or we roll over the columns

Reply:
Stole this from Dask, wasn't sure how far to deviate from logic/naming. Do you have a preference on the preferred name?

Reply:
``_is_blockwise_op`` is better suited

Reply:
Fine to rename in a follow up